From fea3b81c010080a7a3fbb4ce528f36c42d00a02b Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 2 Feb 2016 21:42:45 +0200
Subject: [PATCH 0779/1240] fix: i2c: dts: enable I2C for AP in Armada-7040RZ
 board

- Enable I2C for AP in Armada-7040RZ board
- Fix I2C node in CP-110 (add clock reference for i2c)

Change-Id: I4c83c815cf32acb77540e93d6a39f9cf5de9d158
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27161
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/dts/armada-7040-rz-db_dop.dts | 2 +-
 arch/arm/dts/cp110.dtsi                | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/armada-7040-rz-db_dop.dts b/arch/arm/dts/armada-7040-rz-db_dop.dts
index 506cd21..81fb2f1 100644
--- a/arch/arm/dts/armada-7040-rz-db_dop.dts
+++ b/arch/arm/dts/armada-7040-rz-db_dop.dts
@@ -37,7 +37,7 @@
 				status = "disabled";
 			};
 			i2c0: i2c@511000 {
-				status = "disabled";
+				status = "okay";
 			};
 		};
 		pcie-controller {
diff --git a/arch/arm/dts/cp110.dtsi b/arch/arm/dts/cp110.dtsi
index 34d4374..e0a47a5 100644
--- a/arch/arm/dts/cp110.dtsi
+++ b/arch/arm/dts/cp110.dtsi
@@ -47,11 +47,13 @@ spi1@700680 {
 i2c@701000 { /* i2c0 */
 	compatible = "marvell,mvebu-i2c";
 	reg = <0x701000 0x100>;
+	clock = <&tclk>;
 	status = "disabled";
 };
 i2c@701100 { /* i2c1 */
 	compatible = "marvell,mvebu-i2c";
 	reg = <0x701100 0x100>;
+	clock = <&tclk>;
 	status = "disabled";
 };
 sata@550000 { /* SATA0 */
-- 
1.9.1

